src/main/kotlin/pt/up/fe/ni/website/backend/service/AuthService.kt

```diff
@@ -0,0 +1,89 @@
+package pt.up.fe.ni.website.backend.service
+
+import org.springframework.http.HttpStatus
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.GrantedAuthority
+import org.springframework.security.core.authority.SimpleGrantedAuthority
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.security.crypto.password.PasswordEncoder
+import org.springframework.security.oauth2.jwt.JwtClaimsSet
+import org.springframework.security.oauth2.jwt.JwtDecoder
+import org.springframework.security.oauth2.jwt.JwtEncoder
+import org.springframework.security.oauth2.jwt.JwtEncoderParameters
+import org.springframework.stereotype.Service
+import org.springframework.web.server.ResponseStatusException
+import pt.up.fe.ni.website.backend.config.auth.AuthConfigProperties
+import pt.up.fe.ni.website.backend.model.Account
+import java.time.Duration
+import java.time.Instant
+import java.util.stream.Collectors
+
+@Service
+class AuthService(
+    val accountService: AccountService,
+    val authConfigProperties: AuthConfigProperties,
+    val jwtEncoder: JwtEncoder,
+    val jwtDecoder: JwtDecoder,
+    private val passwordEncoder: PasswordEncoder
+) {
+    fun authenticate(email: String, password: String): Account {
+        val account = accountService.getAccountByEmail(email)
+        if (!passwordEncoder.matches(password, account.password)) {
+            throw ResponseStatusException(HttpStatus.UNPROCESSABLE_ENTITY, "invalid credentials")
```

Comment:
Once again, a bit picky comment. I feel uneasy specifying a static code in the services. I think the whole point of throwing exceptions in the services and handling them in `ErrorController` is to separate business logic from the responses sent to the client and to make sure the same exception is tied to a single status code.

This is my personal view of the issue but there's some discussion around it (https://www.baeldung.com/exception-handling-for-rest-with-spring). An advantage of using ResponseStatusException is, obviously, reducing code size. I will leave the decision to you but I wanted to point this out